graylog2-server/src/main/java/org/graylog2/cluster/lock/MongoLockService.java

```diff
@@ -89,6 +93,18 @@ public Optional<Lock> lock(@Nonnull String resource) {
         return lock(resource, null);
     }
 
+    @Override
+    public Optional<Lock> lock(@Nonnull String resource, int maxConcurrency) {
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(resource));
+        final long l = collection.countDocuments(eq(FIELD_RESOURCE, resource));
+        if (l < maxConcurrency) {
```

Comment:
There is a unique index on `FIELD_RESOURCE` in that collection, so l can never be > 1

Reply:
Missed that. See below for more discussion.